addons/website_payment/static/src/snippets/s_donation/options.js

```diff
@@ -315,6 +315,38 @@ options.registry.Donation = options.Class.extend({
     },
 });
 
+// Superclass for options that need to disable a button from the snippet overlay
+// TODO: In master Export this class from website.
+const DisableOverlayButtonOption = options.Class.extend({
+    // Disable a button of the snippet overlay
+    disableButton: function (buttonName, message) {
+        // TODO refactor in master
+        const className = 'oe_snippet_' + buttonName;
+        this.$overlay.add(this.$overlay.data('$optionsSection')).on('click', '.' + className, this.preventButton);
+        const $button = this.$overlay.add(this.$overlay.data('$optionsSection')).find('.' + className);
+        $button.attr('title', message).tooltip({delay: 0});
+        // TODO In master: add `o_disabled` but keep actual class.
+        $button.removeClass(className); // Disable the functionnality
```

Comment:
Write code in vanilla JS so that we don't have to adapt that in master.

Reply:
Already this class has lot of comments about refactoring in master, secondly same class is used in website module for s_website_form snippet so whenever we will improve it there, we will export it from there and import it here or wherever required so that we don't have to make changes twice.
Thank you : )